pandas/_libs/tslibs/np_datetime.pxd

```diff
@@ -75,11 +75,13 @@ cdef bint cmp_scalar(int64_t lhs, int64_t rhs, int op) except -1
 
 cdef check_dts_bounds(npy_datetimestruct *dts, NPY_DATETIMEUNIT unit=?)
 
-cdef int64_t dtstruct_to_dt64(npy_datetimestruct* dts) nogil
-
-cdef int64_t pydatetime_to_dt64(datetime val, npy_datetimestruct *dts)
+cdef int64_t pydatetime_to_dt64(
+    datetime val, npy_datetimestruct *dts, NPY_DATETIMEUNIT reso=?
```

Comment:
No defaulting `reso` to `NPY_DATETIMEUNIT.NPY_FR_ns`?

Reply:
we'll soon need to be explicit everywhere this is called anyway